src/iceberg/file_writer.h

```diff
@@ -28,7 +28,9 @@
 
 #include "iceberg/arrow_c_data.h"
 #include "iceberg/file_format.h"
+#include "iceberg/metrics.h"
 #include "iceberg/result.h"
+#include "iceberg/schema.h"
```

Comment:
Can we use forward declaration for Schema? I think it should be included in `iceberg/type_fwd.h` already.